### PR TITLE
[FIX] mail: fixed 403 forbidden error

### PR DESCRIPTION
**Pull request by jpr-odoo**
Description of the issue/feature this PR addresses:
Task ID: https://www.odoo.com/web?#id=2061760&action=327&model=project.task&view_type=form&menu_id=4720

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by seb-odoo** [comment]
As discussed on the task, the target of the PR should not be 12.4: it doesn't exist anymore, and the bug was probably introduced earlier.

Could you please find the first branch where the issue happens and rebase/re-target it?

Thanks!

**Comment by tde-banana-odoo** [comment]
It could be great also to write a test (as minimal as possible). In test_mail you can patch the acces rights to simulate people not having access to the parent message (to avoid having to depend on installed modules / current rules). See for example in test_mail/tests/test_mail_activity.py where we mock check_access_rights .

As mail will probably be cleaned for v14 it is necessary to have corner cases being tested to avoid having the same errors popping again.

And finally I think you can at least target 12, if not 11 indeed.

Thanks !

**Comment by jpr-odoo** [comment]
Hello, @seb-odoo   @tde-banana-odoo  I have done re-target the PR with rebase and as per your suggestion, we have added the test case for that.  Thanks!

File: addons/test_mail/tests/test_mail_message.py [hunk 1]
```diff
@@ -291,6 +293,17 @@ def test_mail_message_access_create_reply(self):
         self.message.write({'partner_ids': [(4, self.user_employee.partner_id.id)]})
         self.env['mail.message'].sudo(self.user_employee).create({'model': 'mail.channel', 'res_id': self.group_private.id, 'body': 'Test', 'parent_id': self.message.id})
 
+    def test_mail_message_portal_user_access_parent_message(self):
+        message = self.test_record.message_post(
+            body='<p>This is First Message</p>', subject='Subject',
+            message_type='comment', subtype='mail.mt_note')
+        with patch.object(MailTestSimple, 'check_access_rights', return_value=True):
```
Comment (seb-odoo) [inline addons/test_mail/tests/test_mail_message.py]: I just don't like that we use a mock here, but I don't really have a better idea.

Comment (tde-banana-odoo) [reply addons/test_mail/tests/test_mail_message.py]: I don't understand. It seems we need a test for people not having access to a message that post a new message based on it for references. I am not sure this test tests that behavior as patching check_access_rights with return value True will enables reading on the message.

Anyway we need to 1/ explicitly check that portal user cannot read the new message, 2/ post a message and check everything ran smoothly with message.message-id present in references.

This test is too much high level to ensure it protects against future modifications.

File: addons/test_mail/tests/test_mail_message.py [hunk 2]
```diff
@@ -291,6 +293,29 @@ def test_mail_message_access_create_reply(self):
         self.message.write({'partner_ids': [(4, self.user_employee.partner_id.id)]})
         self.env['mail.message'].sudo(self.user_employee).create({'model': 'mail.channel', 'res_id': self.group_private.id, 'body': 'Test', 'parent_id': self.message.id})
 
+    def test_mail_message_portal_user_access_parent_message(self):
+        Message = self.test_record.message_post(
```
Comment (seb-odoo) [inline addons/test_mail/tests/test_mail_message.py]: You should use lower case here because it is a record.

**Comment by tde-banana-odoo** [comment]
Hello,

Reads good, indeed ! Thanks for updating.

@robodoo r+

**Pull request closed without merging**